server/test/services/CfJsonDocumentContextTest.java

```diff
@@ -135,6 +136,32 @@ public void hasValueAtPath_returnsFalseForMissingPath() {
     assertThat(data.hasValueAtPath(Path.create("not_here!"))).isFalse();
   }
 
+  @Test
+  public void hasNullValueAtPath_returnsTrueForNull() {
+    CfJsonDocumentContext data = new CfJsonDocumentContext();
+    Path path = Path.create("applicant.horses");
+    data.putLong(path, "");
+
+    assertThat(data.hasNullValueAtPath(path)).isTrue();
+  }
+
+  @Test
+  public void hasNullValueAtPath_returnsFalseForValue() {
+    CfJsonDocumentContext data = new CfJsonDocumentContext();
+    Path path = Path.create("applicant.horses");
+    data.putLong(path, 15);
+
+    assertThat(data.hasNullValueAtPath(path)).isFalse();
+  }
+
+  @Test
+  public void hasNullValueAtPath_throwsForMissingPath() {
+    CfJsonDocumentContext data = new CfJsonDocumentContext();
+
+    assertThatThrownBy(() -> data.hasNullValueAtPath(Path.create("not_here!")))
```

Comment:
nit: remove the `!` so we know that `Path.create` won't throw for an invalid path